File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int product = 1;
        int[] output = new int[nums.length];

        for (int n: nums) {
            product = product * n;
        }

        for (int i = 0; i < nums.length; i ++) {
            if (nums[i] != 0) {
                output[i] = product / nums[i];
            } 
            else {
                int p = 1;

                for (int j = 0; j < nums.length; j ++) {
                    if (j != i) {
                        p = p * nums[j];
                    }
                }

                output[i] = p;
            }
        }

        return output;
    }
}  

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []

        if len(nums) == 1 and nums[0] == target:
            res.append(nums[0])
            return res

        def backtrack(index, current_subset, total):
            if total == target:
                res.append(current_subset.copy())
                return

            if index >= len(nums) or total > target:
                return

            current_subset.append(nums[index])
            backtrack(index, current_subset, total + nums[index])

            current_subset.pop()
            backtrack(index + 1, current_subset, total)
        
        backtrack(0, [], 0)
        return res

